github: Provide more suggestions for info to include for a test failure (#2430)
adoptium/infrastructure: .github/ISSUE_TEMPLATE/testcasefail.md

To make it easy for the infrastructure team to repeat and diagnose, please
answer the following questions:

- test suite/name (e.g, BUILD_LIST, TARGET, CUSTOM_TARGET)?

> - a link into recent https://ci.adoptopenjdk.net/view/Test_functional/ runs.

Reviewer: This isn't always the case (only functional tests would have this link) and they expire. Recommend removing this.

Author: * I am trying to address the specific issues I had when revisiting this issue: https://github.com/adoptium/infrastructure/issues/2153. 
* I am struggling with how to specify that the link should not be: https://ci.adoptopenjdk.net/view/Test_functional/job/Test_openjdk8_hs_special.functional_ppc64_aix/28/ (which, as you say expires), but: https://ci.adoptopenjdk.net/view/Test_functional/job/Test_openjdk8_hs_special.functional_ppc64_aix/ which lists the recent runs of the same test.
* PLEASE! help me out here - all the different ways to specify things makes my head spin (and I am trying to make it easier for all of to re-visit testcases).

Author: * Actually, now I see I only need to strip off (from the example above) the /28/ and the link shows the recent examples - at least for those with access.
* wish I had tried that MUCH earlier - starting at https://ci.adoptopenjdk.net and using the navigation, for the novice I am was cumbersome.
* So, again - I would like to keep this suggestion - but perhaps with an improved help text/hint.

Reviewer: @smlambert I think you had a better path suggestion here perhaps?

Reviewer 2: 2 problems of including a Jenkins link in an issue reported: 
- links with specific build IDs are transient (we typically only keep the last 5 runs)
- functional is only one type of test we run (we have openjdk, perf, system, external as well on this server), so if we do suggest including a link, we might be better to say something like `a link into a job on https://ci.adoptopenjdk.net showing the failure, such as xxxxx`

Strongly prefer to include the Rerun in Grinder link from the Jenkins job into an issue as that remains 'timeless' versus transient.  And strongly recommend that the link be hyperlinked, since it is painfully long as it includes ALL parameters and their values that were setup and used in the original test job.  One could also use the Rerun in Grinder on same machine link, which fills in the LABEL with the hostname of the machine that hosted the test run when it failed.

<img width="404" alt="rerunLinksToInclude" src="https://user-images.githubusercontent.com/2836948/149555606-70b40572-c9bd-4b23-b4e1-85ac797be255.png">

Upcoming work will refine the "Rerun in Grinder link to also set the TARGET equal to the failed targets, as a greater convenience to triagers, see related: https://github.com/adoptium/aqa-tests/issues/3219

Test triagers are advised to right-click and copy that link and paste it into the issue as a hyperlink.  They are also asked to grab any relevant or interesting console output from the failing job (this is helpful for intermittent test failures that are difficult to reproduce).   The xxxx_test_output_yyyyy.tar.gz file archived by the test job (shown in above pic) will also contain core files in the case of a crash, which may in some cases be useful to download and look at when debugging.

> - a link into recent https://ci.adoptopenjdk.net/view/Test_functional/ runs.

Reviewer 3: ```suggestion
- a link into recent `Test_` job on https://ci.adoptopenjdk.net which showed the failure
```

> - Grinder re-run to the test with the failure, if available?

Reviewer: ```suggestion
- **Hyperlink** to the Grinder re-run of the test with the failure, if available.
```

> - **Hyperlink** to the Grinder re-run of the test with the failure, if available.

Reviewer 3: As per Shelley's comment:

```suggestion
- **Hyperlink** to re-run in Grinder: 
```
